bmcs/src/main/scala/akka/stream/alpakka/oracle/bmcs/auth/BmcsSigner.scala

```diff
@@ -0,0 +1,145 @@
+/*
+ * Copyright (C) 2016-2017 Lightbend Inc. <http://www.lightbend.com>
+ */
+package akka.stream.alpakka.oracle.bmcs.auth
+
+import java.nio.charset.StandardCharsets
+import java.security.Signature
+import java.security.interfaces.RSAPrivateKey
+import java.time.Instant
+import java.util.Base64
+
+import akka.http.scaladsl.model.headers._
+import akka.http.scaladsl.model._
+import akka.stream.Materializer
+import akka.stream.alpakka.oracle.bmcs.BmcsException
+
+import scala.concurrent.Future
+
+object BmcsSigner {
```

Comment:
This isn't meant to be public api, right?

```
/**
 * INTERNAL API
 */
@InternalApi private[bmcs] BmcsSigner { ... 
```

Reply:
I am not sure if we should mark this as internal api or internal use only. The signer can be used for signing any request for oracle BMCS. Might be usable in other oracle cloud services in future. 
I plan to add some utility methods to create bucket / delete objects.. so that this is a stand alone solution for most of the common tasks, which will also use this signer. 
I want to use this signer in those utility methods.. without having to add those to the connector?
What are your thoughts, do you think this belongs as internal API?